src/test/java/spoon/test/intercession/IntercessionTest.java

```diff
@@ -166,6 +163,8 @@ public boolean matches(CtMethod<?> element) {
 				});
 
 		for (CtMethod<?> setter : setters) {
+			if (setter.getSimpleName().toLowerCase().contains("metadata")) continue;
```

Comment:
I will change it to equals to setMetadata.

This test was failing because setMetadata return type is not parametrized. Which does not make much sense. I guess it is some test you might have written to have guarantees of a well-typed model api. But setMetadata is not a method from the AST API (as in setBody or setTarget).